applications/RANSApplication/custom_conditions/vms_monolithic_k_based_wall_condition.h

```diff
@@ -0,0 +1,278 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_VMS_MONOLITHIC_K_BASED_WALL_CONDITION_WALL_H_INCLUDED)
+#define KRATOS_VMS_MONOLITHIC_K_BASED_WALL_CONDITION_WALL_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_conditions/monolithic_wall_condition.h"
+
+namespace Kratos
+{
+///@addtogroup RANSApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief
+ *
+ * @tparam TDim       Dimensionality of the condition (2D or 3D)
+ * @tparam TNumNodes  Number of nodes in the condition
+ */
+template <unsigned int TDim, unsigned int TNumNodes = TDim>
```

Comment:
```suggestion
template <unsigned int TDim, unsigned int TNumNodes>
```
I'd avoid default  template arguments in these cases to always state clear which is the geometry instance.